src/types.ts

```diff
@@ -0,0 +1,16 @@
+import { Resolver } from './resolvers'
+
+export interface ResolverInternal<T> extends Resolver<T> {
+  /**
+   * True if this resolver is a value resolver. Used to implicit set the lifetime of a value
+   * resolver to either {@link Lifetime.SCOPED} or {@link Lifetime.SINGLETON} depending on whether
+   * this value is registered to a root or a scope container.
+   */
+  isValue?: boolean
+
+  /**
+   * True if this resolver should be excluded from lifetime leak checking. Used to exclude alias
+   * resolvers since any lifetime issues will be caught in the resolution of the alias target.
+   */
```

Comment:
Let's make this a bit more generic as well. It's perfectly fine for custom resolvers to use this. For example, I have an `asLazy` (not in core because I don't recommend it) that resolves when any method is called on the returned proxy.


```suggestion
  /**
   * True if this resolver should be excluded from lifetime leak checking. Used by resolvers
   * that wish to uphold the anti-leakage contract themselves.
   */
```